### ISSUE TITLE
Further clarification on (partial) resolution

### ISSUE THREAD

**Issue by andrewwhitehead** [decentralized-identity/didwebvh#131]
In the `Read (resolve)` section there is this paragraph:

> On completing the processing and successful verification of all [entries](https://identity.foundation/trustdidweb/#term:entries) in the [DID Log](https://identity.foundation/trustdidweb/#term:did-log), respond to the DID resolution request, including the application of DID query [parameters](https://identity.foundation/trustdidweb/#term:parameters) such as `?versionId=` and `?versionTime=` with the appropriate [DIDDoc](https://identity.foundation/trustdidweb/#term:diddoc) version and content.

While I don't disagree with this there are a couple side effects to consider.

- Is it the rule that a more recent, *invalid* log entry will prevent resolution of an earlier `versionId`/`versionTime`? ie. Users would no longer be able to resolve a previous *valid* entry because the log itself appears corrupt. This seems like a good policy from a security perspective.
- I think it might be also relevant when defining the behaviour of a cache, because the entire log file would always need to be processed/verified instead of needing to account for partial verifications. We would also need to define the expiry of a corrupted log, so that an unauthenticated/unwitnessed attack on a log file could be recovered from.
- The witness verification algorithm would also depend on this decision. It becomes more complicated if parsing of the log file might abort early.

**Comment by brianorwhatever** [comment]
> Is it the rule that a more recent, invalid log entry will prevent resolution of an earlier versionId/versionTime? ie. Users would no longer be able to resolve a previous valid entry because the log itself appears corrupt. This seems like a good policy from a security perspective.

I don't think this should be the behaviour as there is nothing invalid about that previous entry

> I think it might be also relevant when defining the behaviour of a cache, because the entire log file would always need to be processed/verified instead of needing to account for partial verifications. We would also need to define the expiry of a corrupted log, so that an unauthenticated/unwitnessed attack on a log file could be recovered from.

yeah definitely.. however less of an issue if we allow early resolution (previous entries aren't invalidated)

> The witness verification algorithm would also depend on this decision. It becomes more complicated if parsing of the log file might abort early.

Witnesses shouldn't be asked to sign an earlier version in fact we should disallow it. so for them processing should never abort early

**Comment by swcurran** [comment]
Do we need to add any clarifications to the spec. related to these questions?  @andrewwhitehead 's document with an overview of the processing carried out by the Python implementation is helpful in defining these issues. But I don't think they change the specification.

We plan to discuss this and other issues at the next did:webvh Work Item meeting -- Thursday, Feb, 13 at 9:00 Pacific / 18:00 Central Europe -- [agenda and zoom info](https://hackmd.io/k4cIK9vQSlaeg2pdHE51IQ?both#Meeting---13-Feb-2025). Our hope at that meeting is to define anything that needs to change (if anything) before the v1.0 spec.

**Comment by swcurran** [comment]
Addressed by PR #186.

**Comment by swcurran** [comment]
I noticed that in the processing instructions, verification fails currently say "terminate resolution".  Given the intent of this change -- to allow the return of valid DIDDocs even if later entries are invalid, changed those references to "flag the error".
